File: 2428-maximum-sum-of-an-hourglass/2428-maximum-sum-of-an-hourglass.java
class Solution {
    public int maxSum(int[][] grid) {
        int res = 0, currSum = 0;

        int m = grid.length, n = grid[0].length;
        for(int i = 0; i<=(m-3); i++){
            for(int j = 0; j<=(n-3); j++){
                currSum += (grid[i][j] + grid[i][j+1] + grid[i][j+2]);
                int mid = (j+(j+2))/2;
                currSum += grid[i+1][mid];
                int k = i+2;
                currSum += (grid[k][j] + grid[k][j+1] + grid[k][j+2]);
                res = Math.max(currSum, res);
                currSum = 0;
            }
        }

        return res;
    }
}

File: 2428-maximum-sum-of-an-hourglass/2428-maximum-sum-of-an-hourglass.py
class Solution:
    def maxSum(self, grid: List[List[int]]) -> int:
        res, currSum = 0, 0

        m, n = len(grid), len(grid[0])
        for i in range((m-3)+1):
            for j in range((n-3)+1):
                currSum += (grid[i][j] + grid[i][j+1] + grid[i][j+2])
                mid = (j+(j+2))//2
                currSum += grid[i+1][mid]
                currSum += (grid[i+2][j] + grid[i+2][j+1] + grid[i+2][j+2])
                res = max(res, currSum)
                currSum = 0
        
        return res